### PR TITLE
Feature/test api

### PR DESCRIPTION
**Pull request by rherbelleau**
add unit tests on timeside/server.

### CODE REVIEW

File: timeside/server/views.py [hunk 1]
```diff
@@ -112,11 +114,20 @@ class Meta:
 class AnnotationTrackViewSet(UUIDViewSetMixin, viewsets.ModelViewSet):
 
     model = models.AnnotationTrack
-    queryset = model.objects.all()
     serializer_class = serializers.AnnotationTrackSerializer
+    queryset=model.objects.all()
     filterset_class = AnnotationTrackFilter
+    
+    def get_queryset(self):
```
[the remark this answers is not in the document]
Comment (rherbelleau) [reply timeside/server/views.py:121]: yes, i don't know why it was not working the last time but now it works well

File: timeside/server/views.py [hunk 2]
```diff
@@ -112,11 +114,20 @@ class Meta:
 class AnnotationTrackViewSet(UUIDViewSetMixin, viewsets.ModelViewSet):
 
     model = models.AnnotationTrack
-    queryset = model.objects.all()
     serializer_class = serializers.AnnotationTrackSerializer
+    queryset=model.objects.all()
```
[the remark this answers is not in the document]
Comment (rherbelleau) [reply timeside/server/views.py:118]: done

File: timeside/server/tests/test_annotation.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+class TestAnnotation(TimeSideTestServer):
+
+    def test_create_annotation_track(self):
+        user=User.objects.all()[0]
+        len_annotation_track=AnnotationTrack.objects.count()
+        data={
+            "item": self.item_url,
+            "title": "test_create_annotation_track",
+            "author": '/timeside/api/users/'+str(user.username)+'/',
```
[the remark this answers is not in the document]
Comment (rherbelleau) [reply timeside/server/tests/test_annotation.py:12]: done

File: timeside/server/tests/test_analyzer.py [hunk 4]
```diff
@@ -1,33 +1,30 @@
...
+    def analyzer_work_test(self, analysis):
         analysis.test=True
         analysis.save()
 
-        params = {'title':'test_analysis_'+str(analysis_uuid),
+        params = {'title':'test_analysis_'+analysis.title,
                 'description':'',
-                'analysis':'/timeside/api/analysis/'+str(analysis_uuid)+ '/',
-                'item': '/timeside/api/items/' + str(self.sweep_32000.uuid) + '/',
+                'analysis':'/timeside/api/analysis/'+str(analysis.uuid)+ '/',
```
[the remark this answers is not in the document]
Comment (rherbelleau) [reply timeside/server/tests/test_analyzer.py:21]: done

File: timeside/server/tests/test_analysis.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            sub_processor=SubProcessor.objects.get(
+                processor=self.processor2)
+            )
+        analysis_obj.test=True
+        analysis_obj.save()
+        analysis=self.client.get('/timeside/api/analysis/'+str(analysis_obj.uuid)+'/', format='json')
+
+        data['analysis']=analysis.data['url']
+        analysis_track=self.client.put(analysis_track.data['url'], data, format='json')
+        self.assertEqual(analysis_track.data['analysis'],data['analysis'])
```
[the remark this answers is not in the document]
Comment (rherbelleau) [reply timeside/server/tests/test_analysis.py:94]: done

File: timeside/server/serializers.py [hunk 6]
```diff
@@ -181,7 +179,7 @@ def to_representation(self, instance):
         self.nb_pixels = int(request.GET.get('nb_pixels', 1024))
 
         from .utils import get_or_run_proc_result
-        result = get_or_run_proc_result('waveform_analyzer', item=instance)
+        result = get_or_run_proc_result('waveform_analyzer', item=instance, test=True)
```
[the remark this answers is not in the document]
Comment (rherbelleau) [reply timeside/server/serializers.py]: I changed that by adding a test bool on item, now you can avoid celery and worker when asking "timeside/api/items/item_uuid/waveform" or "timeside/api/items/item_uuid/download/flac" if item.test=True.

**Comment by yomguy** [comment]
Thanks @rherbelleau for the fixes, great job!

**Pull request merged**